### PR TITLE
Decoupling Bonfire component name and component name

### PR DESCRIPTION
**Pull request by dscervantes**
Sometimes the component name can be different from the bonfire component name

### CODE REVIEW

**Comment by dscervantes** [comment]
@gbenhaim @ernesgonzalez33 can i get a review on this :)

**Comment by ernesgonzalez33** [comment]
@dscervantes @gbenhaim please correct me if I'm wrong, but I think this is introducing a breaking change.

If we don't add a default value to the pipeline parameter, that would mean that all pipelines using this new pipeline will fail starting for now since they are not using the new parameter.

Do we want to continue supporting COMPONENT_NAME for those that has the same name of the component in Konflux and in clowder? Because I also see here that now the component depends only on this new variable: https://github.com/gbenhaim/cicd-tools/blob/main/bin/parse-snapshot.py#L61

**Comment by ernesgonzalez33** [comment]
> @dscervantes @gbenhaim please correct me if I'm wrong, but I think this is introducing a breaking change.
> 
> If we don't add a default value to the pipeline parameter, that would mean that all pipelines using this new pipeline will fail starting for now since they are not using the new parameter.
> 
> Do we want to continue supporting COMPONENT_NAME for those that has the same name of the component in Konflux and in clowder? Because I also see here that now the component depends only on this new variable: https://github.com/gbenhaim/cicd-tools/blob/main/bin/parse-snapshot.py#L61

Ok, Now I see that it is still taking `component.name` as default if the new variable doesn't exits. That would mean that @dscervantes you should add a default value of `""` to the new parameter in the pipeline

File: pipelines/basic.yaml [hunk 1]
```diff
@@ -54,6 +54,9 @@ spec:
       type: string
       default: rh-artifacts-bucket
       description: Secret with connection details to S3
+    - name: BONFIRE_COMPONENT_NAME
+      type: string
+      description: name of the app-sre component name
```
Comment (ernesgonzalez33) [inline pipelines/basic.yaml:59]: Add a default value of `""` to make the parameter optional

File: tasks/deploy.yaml [hunk 2]
```diff
@@ -29,6 +29,9 @@ spec:
     - name: APP_NAME
       type: string
       description: name of app-sre "application" folder this component lives in
+    - name: BONFIRE_COMPONENT_NAME
+      type: string
+      description: name of the app-sre component name
```
Comment (ernesgonzalez33) [inline tasks/deploy.yaml:34]: Add a default value of "" to make the parameter optional

File: tasks/run-cji-smoke-test.yaml [hunk 3]
```diff
@@ -20,6 +20,9 @@ spec:
     - name: COMPONENT_NAME
       type: string
       description: name of app-sre "resourceTemplate" in deploy.yaml for this component
+    - name: BONFIRE_COMPONENT_NAME
+      type: string
+      description: name of the app-sre component name
```
Comment (ernesgonzalez33) [inline tasks/run-cji-smoke-test.yaml:25]: Add a default value of "" to make the parameter optional

File: tasks/run-cji-smoke-test.yaml
Comment (ernesgonzalez33) [inline tasks/run-cji-smoke-test.yaml:1]: For this to work, the function here needs to be updated as well with the new parameter: https://github.com/gbenhaim/cicd-tools/blob/main/bin/deploy-iqe-cji.sh

**Review by ernesgonzalez33 — CHANGES_REQUESTED** [review]

File: tasks/run-cji-smoke-test.yaml
Comment (ernesgonzalez33) [reply tasks/run-cji-smoke-test.yaml:1]: Being worked on it here: https://github.com/gbenhaim/cicd-tools/pull/15

**Comment by dscervantes** [comment]
@ernesgonzalez33 Can you let me know when the other PR is able to be merged https://github.com/gbenhaim/cicd-tools/pull/15 thanks! fixed the other defaults

**Review by dscervantes — COMMENTED** [review]
done

**Comment by dscervantes** [comment]
@ernesgonzalez33 can I get an approval? It doesn't let me test without it

**Comment by ernesgonzalez33** [comment]
@dscervantes it was already tested. Failed again: https://github.com/gbenhaim/tekton-insights/pull/25/checks?check_run_id=24004374506. I think the issue is cicd-tools. I'll check it today

File: tasks/run-cji-smoke-test.yaml [hunk 4]
```diff
@@ -87,6 +91,8 @@ spec:
               key: url
         - name: COMPONENT_NAME
           value: $(params.COMPONENT_NAME)
+        - name: BONFIRE_COMPONENT_NAME
+          value: $(BONFIRE_COMPONENT_NAME)
```
Comment (ernesgonzalez33) [inline tasks/run-cji-smoke-test.yaml:95]: this should be `$(params.BONFIRE_COMPONENT_NAME)`

**Review by ernesgonzalez33 — CHANGES_REQUESTED** [review]

**Review by ernesgonzalez33 — APPROVED** [review]

**Pull request merged**